butte-examples/fbs/array.fbs

```diff
@@ -0,0 +1,19 @@
+/// Flatbuffers namespaces are defined as UpperCamelCase.
+namespace Array;
+
+/// A test entry with a value.
+table Entry {
+  value: uint64;
+}
+
+enum Value: ubyte {
+  A = 0,
+  B = 1
+}
+
+/// An array of tables.
+table ArrayTable {
+  table_entries: [Entry];
+  enum_entries: [Value];
+  int_entries: [int32];
+}
```

Comment:
Can you add an array-of-array example here and a corresponding assertion in `array.rs`? If that fails for an unrelated reason, we can merge this PR and create a follow up issue with the test case.

Reply:
I rebased the PR and added an extra array of arrays test.

It works fine but please note that this is not supported by flatc: error: nested vector types not supported (wrap in table first)